api/src/main/java/com/datastrato/gravitino/file/File.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.file;
+
+import com.datastrato.gravitino.Auditable;
+import com.datastrato.gravitino.Namespace;
+import java.util.Collections;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * An interface representing a file in a schema {@link Namespace}. It defines the basic properties
+ * of a file object. A catalog implementation with {@link FileCatalog} should implement this
+ * interface.
+ */
+public interface File extends Auditable {
+
+  /** An enum representing the format of the file. */
+  enum Format {
+    CSV,
+    JSON,
+    PARQUET,
+    AVRO,
+    ORC,
+    TEXT,
+    BINARY,
+    IMAGE,
+    TFRECORD,
+    UNKNOWN
+  }
+
+  /** @return Name of the file object. */
+  String name();
+
+  /** @return The comment of the file object. Null is returned if no comment is set. */
+  @Nullable
+  default String comment() {
+    return null;
+  }
+
+  /** @return the format of the underlying filesets that is managed by this file object. */
+  Format format();
```

Comment:
Should we set the format for the `File`? If we did, it will assume all files in a `File` must be the same format.
is there possible different file format on different directory?

For example, we use orc as our current file format, but one day we want to change it to parquet. What should I do with the history files?